evm/rpc/modules/eth.py

```diff
@@ -1,15 +1,40 @@
+from contextlib import contextmanager
+from cytoolz import (
+    identity,
+)
+
 from eth_utils import (
+    apply_to_return_value,
     decode_hex,
 )
 
 from evm.rpc.format import (
     block_to_dict,
+    format_params,
+    hex_to_int,
 )
 from evm.rpc.modules import (
     RPCModule,
 )
 
 
+@contextmanager
+def state_at_block(chain, at_block, read_only=True):
+    if at_block == 'pending':
+        at_header = chain.header
+    elif at_block == 'latest':
+        at_header = chain.get_canonical_head()
+    elif at_block == 'earliest':
+        # TODO find if genesis block can be non-zero. Why does 'earliest' option even exist?
+        at_header = chain.get_canonical_block_by_number(0).header
+    else:
```

Comment:
We can potentially avoid some confusing errors down the line by making this an `elif is_hex_or_int(at_block):` and then putting an exception in the final `else` statement so that we are *sure* that `at_block` is well formed.